tests/gpflow/experimental/check_shapes/test_integration.py

```diff
@@ -115,58 +115,108 @@ def test_layer(join: SuperLayer) -> None:
     test_layer(SubLayer())
 
 
-_F = Callable[[tf.Tensor], tf.Tensor]
+_Err = Callable[[tf.Tensor, tf.Tensor], tf.Tensor]
 _Loss = Callable[[], tf.Tensor]
 
+_IdWrapper = lambda x: x
+_TfFunction = tf.function
+_ShapedTfFunctionErr = tf.function(
+    input_signature=[
+        tf.TensorSpec(shape=[None], dtype=tf.float64),
+        tf.TensorSpec(shape=[], dtype=tf.float64),
+    ]
+)
+
+_ShapedTfFunctionLoss = tf.function(input_signature=[])
+_UnshapedTfFunctionErr = tf.function(
+    input_signature=[
+        tf.TensorSpec(shape=None, dtype=tf.float64),
+        tf.TensorSpec(shape=None, dtype=tf.float64),
+    ]
+)
+_UnshapedTfFunctionLoss = tf.function(input_signature=[])
+_RelaxedTfFunction = tf.function(experimental_relax_shapes=True)
+
+_NoneShape = None
+_TargetShape = tf.TensorShape([])
+_VShape = tf.TensorShape([50])
+_UnknownShape = tf.TensorShape(None)
+
 
 @pytest.mark.parametrize(
-    "f_wrapper,loss_wrapper",
+    "err_wrapper,loss_wrapper",
     [
-        (
-            lambda x: x,
-            lambda x: x,
-        ),
-        (
-            tf.function,
-            tf.function,
-        ),
-        (
-            tf.function(input_signature=[tf.TensorSpec(shape=[None], dtype=tf.float64)]),
-            tf.function(input_signature=[]),
-        ),
-        (
-            tf.function(input_signature=[tf.TensorSpec(shape=None, dtype=tf.float64)]),
-            tf.function(input_signature=[]),
-        ),
-        (
-            # See: https://github.com/tensorflow/tensorflow/issues/56414
-            tf.function(experimental_relax_shapes=Version(tf.__version__) < Version("2.9.0")),
-            tf.function(experimental_relax_shapes=True),
-        ),
+        (_IdWrapper, _IdWrapper),
+        (_TfFunction, _TfFunction),
+        (_ShapedTfFunctionErr, _ShapedTfFunctionLoss),
```

Comment:
If you're always testing identical error and loss wrappers, would it be better to have just one parameter?

Reply:
> If you're always testing identical error and loss wrappers, would it be better to have just one parameter?

But they're not always the same. `_ShapedTfFunctionErr` is not  `_ShapedTfFunctionLoss`.